be/src/olap/base_compaction.cpp

```diff
@@ -16,513 +16,115 @@
-
-        for (vector<RowsetSharedPtr>::iterator it = unused_indices->begin();
-                it != unused_indices->end(); ++it) {
-            storage_engine->add_unused_rowset(*it);
-        }
-    }
-}
-
-void BaseCompaction::_garbage_collection() {
-    // 清理掉已生成的版本文件
-    StorageEngine* storage_engine = StorageEngine::instance();
-    for (vector<RowsetSharedPtr>::iterator it = _new_rowsets.begin();
-            it != _new_rowsets.end(); ++it) {
-        storage_engine->add_unused_rowset(*it);
+                  << ", interval_since_last_base_compaction=" << interval_since_last_base_compaction 
+                   << ", interval_threshold=" << interval_threshold;
+        return OLAP_SUCCESS;
     }
-    _new_rowsets.clear();
 
-    _release_base_compaction_lock();
+    LOG(INFO) << "don't satisfy the base compaction policy. tablet=" << _tablet->full_name()
+              << ", num_cumulative_rowsets=" << _input_rowsets.size() - 1
+              << ", cumulative_base_ratio=" << cumulative_base_ratio
+              << ", interval_since_last_base_compaction=" << interval_since_last_base_compaction;
+    return OLAP_ERR_BE_NO_SUITABLE_VERSION;
 }
 
-bool BaseCompaction::_validate_need_merged_versions(
-        const vector<Version>& candidate_versions) {
-    if (candidate_versions.size() <= 1) {
-        LOG(WARNING) << "unenough versions need to be merged. size=" << candidate_versions.size();
-        return false;
-    }
-
-    // 1. validate versions in candidate_versions are continuous
-    // Skip the first element
-    for (unsigned int index = 1; index < candidate_versions.size(); ++index) {
-        Version previous_version = candidate_versions[index - 1];
-        Version current_version = candidate_versions[index];
-        if (current_version.first != previous_version.second + 1) {
-            LOG(WARNING) << "wrong need merged version. "
-                         << "previous_version=" << previous_version.first
-                         << "-" << previous_version.second
-                         << ", current_version=" << current_version.first
-                         << "-" << current_version.second;
-            return false;
-        }
-    }
-
-    // 2. validate m_new_base_version is OK
-    if (_new_base_version.first != 0
-            || _new_base_version.first != candidate_versions.begin()->first
-            || _new_base_version.second != candidate_versions.rbegin()->second) {
-        LOG(WARNING) << "new_base_version is wrong."
-                     << " new_base_version=" << _new_base_version.first
-                     << "-" << _new_base_version.second
-                     << ", vector_version=" << candidate_versions.begin()->first
-                     << "-" << candidate_versions.rbegin()->second;
-        return false;
-    }
-
-    VLOG(10) << "valid need merged version";
-    return true;
+const std::string& BaseCompaction::compaction_name() const {
+    return _compaction_name;
```

Comment:
Just return const value too

Reply:
I want to use reference to reduce string copy.
If use const value, it will cause string copy.